src/main/java/uk/gov/hmcts/reform/cpo/security/ServicePermissionInfo.java

```diff
@@ -0,0 +1,14 @@
+package uk.gov.hmcts.reform.cpo.security;
+
+import lombok.Data;
+import uk.gov.hmcts.reform.cpo.validators.ValidationError;
+
+import javax.validation.constraints.Pattern;
+
+@Data
+class ServicePermissionInfo {
+    private String id;
+
+    @Pattern(regexp = "^[crudCRUD]+", message = ValidationError.INVALID_PERMISSION_WHITELIST_VALUE)
```

Comment:
can we move this class as static inner one in ServiceAuthorizationConfig ?

Reply:
I did try that, but without much luck - will take another look

Reply:
Still no luck, compiles but tests fail as 

`public void setAuthorizations(Map<String, @Valid ServicePermissionInfo> authorizations)`

in ServiceAuthorizationConfig doesn't get called.

Will leave as is for now

Reply:
ok, np. Thanks for trying.